src/Endpoints/class-base-api-proxy.php

```diff
@@ -172,4 +174,89 @@ protected function get_data( WP_REST_Request $request, bool $require_api_secret
 			'data' => $this->generate_data( $response ), // @phpstan-ignore-line.
 		);
 	}
+
+	/**
+	 * Extracts the post data from the passed object.
+	 *
+	 * Should only be used with endpoints that return post data.
+	 *
+	 * @since 3.10.0
+	 *
+	 * @param stdClass $item The object to extract the data from.
+	 *
+	 * @return array<string, mixed> The extracted data.
+	 */
+	protected function extract_post_data( stdClass $item ): array {
```

Comment:
Can we please confirm with Parse.ly Backend team that for both endpoints they are using same code for generating post data object?

- If YES then we can keep this change (although I do prefer to map API responses separately to be cautious and it's also helpful to only have the needed fields after mapping)

- If NO then I think we should also avoid doing this on our side

Reply:
Getting an answer to that could take some time, which is of the essence right now. Regardless, implementations that are out of our control can change without notice. So today it could be the same code, tomorrow it could differ. This shouldn't make any difference for us in our function(s).

If an (unlikely) code change on the Parse.ly back-end results in breaking our implementation (for example if a field was moved or renamed), it would have broken the old implementation as well. So we would end up with the exact same problem, whether we use one or multiple functions.

I prefer a single function that can be flexible rather than code duplication, especially if more `/post*` endpoints get added in the future.

However, if you strongly disagree with the approach here, we can ignore this PR for the time being. However I'll need to know that to update the code to work without this approach.

Reply:
Let's take another opinion cc: @mjangda